rfd/0097-teleport-connect-usage-metrics.md

```diff
@@ -0,0 +1,179 @@
+may change) and before closing the app.
+
+It was considered to use an authorized endpoint provided by cluster's `Auth Server`, but it seems to not work well for
+Connect for a few reasons:
+
+- Some events may not belong to any cluster (at the time of writing this RFD there is no such event, but the solution
+  should be future-proof).
+- Batch can contain events from multiple clusters.
+- Batch can be sent after the session expires.
+
+### Anonymization
+
+> **NOTE:** The anonymization solution described below applies only to events that are associated with a cluster. Events
+> that do not belong to any cluster but contain sensitive data will have to be anonymized in a different way.
+
+Each event that contains sensitive data, like cluster name needs to be anonymized. It will be done in `tsh daemon`, the
+same way as in `Auth Server` - using HMAC with unique `cluster id` as the key. Connect will reuse the same code.
+The only issue with anonymizing events client-side is lack of `cluster id` that is kept in `Auth Server`. To remedy
+this, when the app starts and retrieves cluster information, it should also retrieve the `cluster id`, create an
+anonymizer and store it in the cluster struct.
+
+### Storing events
+
+Batches of anonymized events will be sent to a public endpoint in `prehog` (intended for use only by Connect) that
+translates them into the PostHog's data model.
+
+Connect events will share the same project with clusters and website events. It will allow to perform queries that need
+both sources of data, like calculating what is the percentage of users logging to a cluster with Connect.
+
+Some event properties, like OS can be saved as a user property. These properties are
+then [stored directly on each event](https://posthog.com/docs/how-posthog-works/data-model#person). For example, when
+the first emitted event sets a user property `os: windows`, each next event will have this property set.
+
+To differentiate events coming from multiple application instances, each event needs to have `distinct_id` field. It
+will be supplied with UUID generated by Connect with `connect.` prefix. The value will be created on the start and
+stored in a file in the app data directory, so it will not change between restarts.
+
+> **NOTE:** As stated above, Connect events are tied to the application instance (or just the client machine). It means
+> that PostHog's _Person_ for Connect and for cluster will be a different thing. They should not be merged.
+
+### User agreement
+
+On the start, Teleport Connect will ask user to opt in to volunteer anonymized metrics and usage-data with standard
+message "Are you OK sending anonymized usage data about Teleport Connect? This will help us to improve product".
+
+If the user refuses, Connect will not send any usage data.
+
+### How will collecting metrics support product development?
+
+In the initial version, it should help with getting answers to the following questions:
```

Comment:
@gzdunek one additional question we should answer is "What persona is using Teleport Connect?"

To answer this question, I think we should ask users to optionally input their role. I think we can give them a few checkboxes for common roles we can think of and give them a custom text input field to enter custom answers if they wish.

Common roles imo:
- Software Engineer
- Support Engineer
- DevOps Engineer
- Solutions Architect
- System Administrator